### PR TITLE
access: removal of external authentications

### LINKED ISSUE #1083: access: remove legacy external authentication methods
Originally

The following config variables:

- from `access_control_config.py`:
```python

# List of tags containing (multiple) emails of users who should authorize
# to access the corresponding record regardless of collection restrictions.
if CFG_CERN_SITE:
    CFG_ACC_GRANT_AUTHOR_RIGHTS_TO_EMAILS_IN_TAGS = ['859__f', '270__m', '506__m']
else:
    CFG_ACC_GRANT_AUTHOR_RIGHTS_TO_EMAILS_IN_TAGS = ['8560_f']


# CFG_EXTERNAL_AUTHENTICATION -- this is a dictionary with the enabled login method.
# The key is the name of the login method and the value is an instance of
# of the login method (see /help/admin/webaccess-admin-guide#5). Set the value
# to None if you wish to use the local Invenio authentication method.
# CFG_EXTERNAL_AUTH_DEFAULT -- set this to the key in CFG_EXTERNAL_AUTHENTICATION
# that should be considered as default login method
# CFG_EXTERNAL_AUTH_USING_SSO -- set this to the login method name of an SSO
# login method, if any, otherwise set this to None.
# CFG_EXTERNAL_AUTH_LOGOUT_SSO -- if CFG_EXTERNAL_AUTH_USING_SSO was not None
# set this to the URL that should be contacted to perform an SSO logout
```

- from `external_authentication_sso.py`
```python

# Tunable list of settings to be hidden
CFG_EXTERNAL_AUTH_HIDDEN_SETTINGS = ('auth', 'respccid', 'personid')
# Tunable list of groups to be hidden
CFG_EXTERNAL_AUTH_HIDDEN_GROUPS = (
...)

CFG_EXTERNAL_AUTH_HIDDEN_GROUPS_RE = (
...)


# Prefix name for Shibboleth variables
CFG_EXTERNAL_AUTH_SSO_PREFIX_NAME = 'ADFS_'
# Name of the variable containing groups
CFG_EXTERNAL_AUTH_SSO_GROUP_VARIABLE = CFG_EXTERNAL_AUTH_SSO_PREFIX_NAME+'GROUP'
# Name of the variable containing login name
CFG_EXTERNAL_AUTH_SSO_LOGIN_VARIABLE = CFG_EXTERNAL_AUTH_SSO_PREFIX_NAME+'LOGIN'
# Name of the variable containing email
CFG_EXTERNAL_AUTH_SSO_EMAIL_VARIABLE = CFG_EXTERNAL_AUTH_SSO_PREFIX_NAME+'EMAIL'
# Name of the variable containing groups
CFG_EXTERNAL_AUTH_SSO_GROUP_VARIABLE = CFG_EXTERNAL_AUTH_SSO_PREFIX_NAME+'GROUP'
# Separator character for group variable
CFG_EXTERNAL_AUTH_SSO_GROUPS_SEPARATOR = ';'
```

- from `external_authentication_ldap.py`
```python

CFG_EXTERNAL_AUTH_LDAP_SERVERS = ['ldap://scoldap.epfl.ch']
CFG_EXTERNAL_AUTH_LDAP_CONTEXT = "o=EPFL,c=CH"
CFG_EXTERNAL_AUTH_LDAP_USER_UID  = ["uid", "uniqueIdentifier", "mail"]
CFG_EXTERNAL_AUTH_LDAP_MAIL_ENTRY = 'mail'
CFG_EXTERNAL_AUTH_LDAP_GROUP_MEMBERSHIP = 'memberOf'
CFG_EXTERNAL_AUTH_LDAP_GROUP_UID = 'uniqueIdentifier'
CFG_EXTERNAL_AUTH_LDAP_GROUP_NAME = 'displayName'

CFG_EXTERNAL_AUTH_LDAP_HIDDEN_GROUPS = ['EPFL-unit', 'users']
```

should be all moved into `invenio.conf` so that that the whole authentication system could be decoupled from CERN/INSPIRE/OpenAIRE/EPFL specific configurations.

### LINKED ISSUE #2972: RFC removal of robotlogin
Robot login is a proprietary authentication protocol (similar to OAUTH), initially developed for the *OpenAIRE Orphan Repository* to exploit the *OpenAIRE Portal* for authentication, and later adopted by the *INSPIRE* service to exploit *arXiv.org* for authentication.

Today the OpenAIRE Orphan Repository has been replaced by Zenodo, and INSPIRE is little by little upgraded to INSPIRE Labs which sports native authentication (OAUTH-based) from ORCID.

Therefore I'd propose to deprecate it.

INSPIRE will be able to still import data from arXiv using a dedicated module.

### PR DESCRIPTION
**Pull request by jirikuncar**
* INCOMPATIBILITY Removes proprietary authentication protocol for
  robotlogin.  (closes #2972)

* INCOMPATIBILITY Removes external authentication engines. Please use
  `invenio.modules.oauthclient` or Flask-SSO instead.  (closes #1083)

Signed-off-by: Jiri Kuncar <jiri.kuncar@cern.ch>

### CODE REVIEW

**Comment by jirikuncar** [comment]
@kaplun can you have a look and let me know if there are some other places to be amended. Thanks

File: invenio/ext/legacy/layout.py [hunk 1]
```diff
@@ -181,7 +181,6 @@ class WebInterfaceInvenio(WebInterfaceSearchInterfacePages):
                    'yourloans',
                    'yourcomments',
                    'ill',
-                   'yourgroups',
```
Comment (lnielsen) [inline invenio/ext/legacy/layout.py]: Is this supposed to be removed in this commit?

Comment (jirikuncar) [inline invenio/ext/legacy/layout.py]: I'm removing dead code here. It *should* be fine.

**Comment by jirikuncar** [comment]
@jalavik @lnielsen can you please have a look?

**Pull request merged**